### PR TITLE
feat: update libp2p

### PR DESCRIPTION
**Pull request by wemeetagain**
- update libp2p to v1.x (and associated dependencies)
  - our noise crypto optimizations have been upstreamed and are no longer needed
  - discv5 is now split into two packages: enr and discv5 (for rationale see https://github.com/ChainSafe/discv5/pull/267)
- revamp libp2p dashboard with relevant metrics
- deployed to feat3 on Jan 11

### CODE REVIEW

File: packages/beacon-node/package.json [hunk 1]
```diff
@@ -99,9 +99,10 @@
     "@chainsafe/as-sha256": "^0.3.1",
     "@chainsafe/bls": "7.1.1",
     "@chainsafe/blst": "^0.2.9",
-    "@chainsafe/discv5": "^5.1.0",
-    "@chainsafe/libp2p-gossipsub": "^10.1.1",
-    "@chainsafe/libp2p-noise": "^13.0.1",
+    "@chainsafe/discv5": "^7.1.0",
+    "@chainsafe/enr": "^2.0.2",
+    "@chainsafe/libp2p-gossipsub": "^11.0.1",
```
Comment (twoeths) [inline packages/beacon-node/package.json]: @wemeetagain  is it worth to use v11.1.0 for this upgrade? that's for `batchPublish` option https://github.com/ChainSafe/js-libp2p-gossipsub/pull/482

Comment (wemeetagain) [reply packages/beacon-node/package.json]: Yeah, we can upgrade it to 11.1.0. Should we turn on batchPublish in certain places?

File: packages/beacon-node/src/network/libp2p/index.ts [hunk 2]
```diff
@@ -91,14 +90,15 @@ export async function createNodeJsLibp2p(
       ? prometheusMetrics({
           collectDefaultMetrics: false,
           preserveExistingMetrics: true,
-          registry: nodeJsLibp2pOpts.metricsRegistry,
+          // TODO fix type here
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-explicit-any
+          registry: nodeJsLibp2pOpts.metricsRegistry as any,
```
Comment (nflaig) [inline packages/beacon-node/src/network/libp2p/index.ts]: This should not be required types seem to have exact match, it is likely when you started working on this branch that we had an older prom version in Lodestar but it got updated in  https://github.com/ChainSafe/lodestar/pull/6230.

```suggestion
          registry: nodeJsLibp2pOpts.metricsRegistry,
```

Comment (nflaig) [reply packages/beacon-node/src/network/libp2p/index.ts]: Cleaned up the type casts and also removed temporary workaround for discv5 metrics added in https://github.com/ChainSafe/lodestar/pull/6201

File: packages/beacon-node/package.json (hunk 1 above)
Comment (twoeths) [reply packages/beacon-node/package.json]: yes just need to add a new option to `Eth2Gossipsub` constructor `batchPublish: true`. That PR need to release first

**Comment by twoeths** [comment]
I see some great metrics with this PR (mostly on the test mainnet node)
- attnet count with >0 mesh peers 
- Network Worker Message Latency
- Dial Error Rate
- `gc` on both network thread and main thread
- memory on network worker (especially lg1k) tends to be more stable but not sure in the long run

looks great overall 🚀

**Review by twoeths — DISMISSED** [review]

**Review by nflaig — DISMISSED** [review]

File: packages/beacon-node/package.json (hunk 1 above)
Comment (twoeths) [reply packages/beacon-node/package.json]: found some weird metrics with `batchPublish` option so let's not do it in this PR. "Validator miss ratio" is so high and balance delta is so high too (first half is with `batchPublish` as true and 2nd half is false)

Screenshot 2024-01-16 at 20 19 40

**Review by nflaig — APPROVED** [review]

**Review by nazarhussain — APPROVED** [review]

**Review by g11tech — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.15.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.15.0) :tada:

**Pull request merged**